### ISSUE TITLE
scriptnum_tests/creation fails on macOS

### ISSUE THREAD

**Issue by rex4539** [zcash/zcash#4458]
Latest master.

```
/Users/rex/zcash on branch master rex@MacBook-Pro-2018% lldb -- src/test/test_bitcoin -t scriptnum_tests/creation
(lldb) target create "src/test/test_bitcoin"
Current executable set to '/Users/rex/zcash/src/test/test_bitcoin' (x86_64).
(lldb) settings set -- target.run-args  "-t" "scriptnum_tests/creation"
(lldb) r
Process 35013 launched: '/Users/rex/zcash/src/test/test_bitcoin' (x86_64)
test_bitcoin(35013,0x1039dadc0) malloc: adding guard pages for large allocator blocks
test_bitcoin(35013,0x1039dadc0) malloc: enabling scribbling to detect mods to free blocks
Running 1 test case...
test_bitcoin was compiled with optimization - stepping may behave oddly; variables may not be available.
Process 35013 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_INSTRUCTION (code=EXC_I386_INVOP, subcode=0x0)
    frame #0: 0x000000010025c481 test_bitcoin`scriptnum_tests::creation::test_method(this=<unavailable>) at scriptnum_tests.cpp:171:33 [opt]
   168 	        for(size_t j = 0; j < sizeof(offsets) / sizeof(offsets[0]); ++j)
   169 	        {
   170 	            RunCreate(values[i]);
-> 171 	            RunCreate(values[i] + offsets[j]);
   172 	            RunCreate(values[i] - offsets[j]);
   173 	        }
   174 	    }
Target 0: (test_bitcoin) stopped.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_INSTRUCTION (code=EXC_I386_INVOP, subcode=0x0)
  * frame #0: 0x000000010025c481 test_bitcoin`scriptnum_tests::creation::test_method(this=<unavailable>) at scriptnum_tests.cpp:171:33 [opt]
    frame #1: 0x000000010025c1c0 test_bitcoin`scriptnum_tests::creation_invoker() at scriptnum_tests.cpp:164:1 [opt]
    frame #2: 0x0000000100aaa99f test_bitcoin`boost::detail::function::function_obj_invoker0<boost::detail::forward, int>::invoke(boost::detail::function::function_buffer&) [inlined] boost::function0<void>::operator(this=<unavailable>)() const at function_template.hpp:763:14 [opt]
    frame #3: 0x0000000100aaa98c test_bitcoin`boost::detail::function::function_obj_invoker0<boost::detail::forward, int>::invoke(boost::detail::function::function_buffer&) [inlined] boost::detail::forward::operator(this=<unavailable>)() at execution_monitor.ipp:1368 [opt]
    frame #4: 0x0000000100aaa984 test_bitcoin`boost::detail::function::function_obj_invoker0<boost::detail::forward, int>::invoke(function_obj_ptr=<unavailable>) at function_template.hpp:137 [opt]
    frame #5: 0x0000000100aa76d6 test_bitcoin`boost::execution_monitor::catch_signals(boost::function<int ()> const&) [inlined] boost::function0<int>::operator(this=<unavailable>)() const at function_template.hpp:763:14 [opt]
    frame #6: 0x0000000100aa76c0 test_bitcoin`boost::execution_monitor::catch_signals(boost::function<int ()> const&) [inlined] int boost::detail::do_invoke<boost::shared_ptr<boost::detail::translator_holder_base>, boost::function<int ()> >(tr=<unavailable>, F=<unavailable>)> const&) at execution_monitor.ipp:290 [opt]
    frame #7: 0x0000000100aa76ac test_bitcoin`boost::execution_monitor::catch_signals(this=0x000000010180c700, F=<unavailable>)> const&) at execution_monitor.ipp:879 [opt]
    frame #8: 0x0000000100aa784c test_bitcoin`boost::execution_monitor::execute(this=0x000000010180c700, F=0x00007ffeefbfec70)> const&) at execution_monitor.ipp:1277:16 [opt]
    frame #9: 0x0000000100aa934b test_bitcoin`boost::execution_monitor::vexecute(this=<unavailable>, F=<unavailable>)> const&) at execution_monitor.ipp:1377:5 [opt]
    frame #10: 0x0000000100aca6fc test_bitcoin`boost::unit_test::unit_test_monitor_t::execute_and_translate(this=0x000000010180c700, func=0x0000000105011318, timeout_microseconds=<unavailable>)> const&, unsigned long) at unit_test_monitor.ipp:49:9 [opt]
    frame #11: 0x0000000100ab3759 test_bitcoin`boost::unit_test::framework::state::execute_test_tree(this=0x000000010180c478, tu_id=<unavailable>, timeout_microseconds=<unavailable>, p_random_generator=<unavailable>) at framework.ipp:823:44 [opt]
    frame #12: 0x0000000100ab42dc test_bitcoin`boost::unit_test::framework::state::execute_test_tree(this=0x000000010180c478, tu_id=<unavailable>, timeout_microseconds=0, p_random_generator=<unavailable>) at framework.ipp:744:54 [opt]
    frame #13: 0x0000000100ab42dc test_bitcoin`boost::unit_test::framework::state::execute_test_tree(this=0x000000010180c478, tu_id=<unavailable>, timeout_microseconds=0, p_random_generator=<unavailable>) at framework.ipp:744:54 [opt]
    frame #14: 0x0000000100ab1c1d test_bitcoin`boost::unit_test::framework::run(id=1, continue_test=<unavailable>) at framework.ipp:1688:29 [opt]
    frame #15: 0x0000000100ac983a test_bitcoin`boost::unit_test::unit_test_main(init_func=<unavailable>, argc=<unavailable>, argv=<unavailable>)(int, char**), int, char**) at unit_test_main.ipp:248:9 [opt]
    frame #16: 0x00007fff6a643cc9 libdyld.dylib`start + 1
(lldb) disassemble --frame
test_bitcoin`scriptnum_tests::creation::test_method:
    0x10025c3c0 <+0>:   pushq  %rbp
    0x10025c3c1 <+1>:   movq   %rsp, %rbp
    0x10025c3c4 <+4>:   pushq  %r15
    0x10025c3c6 <+6>:   pushq  %r14
    0x10025c3c8 <+8>:   pushq  %r13
    0x10025c3ca <+10>:  pushq  %r12
    0x10025c3cc <+12>:  pushq  %rbx
    0x10025c3cd <+13>:  subq   $0x28, %rsp
    0x10025c3d1 <+17>:  movq   0x1575730(%rip), %rax     ; (void *)0x00007fff90899d40: __stack_chk_guard
    0x10025c3d8 <+24>:  movq   (%rax), %rax
    0x10025c3db <+27>:  movq   %rax, -0x30(%rbp)
    0x10025c3df <+31>:  xorl   %eax, %eax
    0x10025c3e1 <+33>:  leaq   -0x38(%rbp), %rbx
    0x10025c3e5 <+37>:  nopw   %cs:(%rax,%rax)
    0x10025c3ef <+47>:  nop    
    0x10025c3f0 <+48>:  movq   %rax, -0x40(%rbp)
    0x10025c3f4 <+52>:  leaq   0xc29e05(%rip), %rcx      ; scriptnum_tests::values
    0x10025c3fb <+59>:  leaq   (%rcx,%rax,8), %rax
    0x10025c3ff <+63>:  movq   %rax, -0x48(%rbp)
    0x10025c403 <+67>:  leaq   0xc29e56(%rip), %r13      ; scriptnum_tests::offsets
    0x10025c40a <+74>:  xorl   %r12d, %r12d
    0x10025c40d <+77>:  nopl   (%rax)
    0x10025c410 <+80>:  movq   -0x48(%rbp), %r14
    0x10025c414 <+84>:  movq   %r14, %rdi
    0x10025c417 <+87>:  callq  0x10025c490               ; scriptnum_tests::RunCreate at scriptnum_tests.cpp:145
    0x10025c41c <+92>:  movq   (%r14), %r15
    0x10025c41f <+95>:  movq   (%r13), %r14
    0x10025c423 <+99>:  movq   %r15, %rax
    0x10025c426 <+102>: addq   %r14, %rax
    0x10025c429 <+105>: jo     0x10025c481               ; <+193> at scriptnum_tests.cpp:171:33
    0x10025c42b <+107>: movq   %rax, -0x38(%rbp)
    0x10025c42f <+111>: movq   %rbx, %rdi
    0x10025c432 <+114>: callq  0x10025c490               ; scriptnum_tests::RunCreate at scriptnum_tests.cpp:145
    0x10025c437 <+119>: subq   %r14, %r15
    0x10025c43a <+122>: jo     0x10025c481               ; <+193> at scriptnum_tests.cpp:171:33
    0x10025c43c <+124>: movq   %r15, -0x38(%rbp)
    0x10025c440 <+128>: movq   %rbx, %rdi
    0x10025c443 <+131>: callq  0x10025c490               ; scriptnum_tests::RunCreate at scriptnum_tests.cpp:145
    0x10025c448 <+136>: incq   %r12
    0x10025c44b <+139>: addq   $0x8, %r13
    0x10025c44f <+143>: cmpq   $0x9, %r12
    0x10025c453 <+147>: jb     0x10025c410               ; <+80> at scriptnum_tests.cpp
    0x10025c455 <+149>: movq   -0x40(%rbp), %rax
    0x10025c459 <+153>: incq   %rax
    0x10025c45c <+156>: cmpq   $0xc, %rax
    0x10025c460 <+160>: jb     0x10025c3f0               ; <+48> at scriptnum_tests.cpp:165
    0x10025c462 <+162>: movq   0x157569f(%rip), %rax     ; (void *)0x00007fff90899d40: __stack_chk_guard
    0x10025c469 <+169>: movq   (%rax), %rax
    0x10025c46c <+172>: cmpq   -0x30(%rbp), %rax
    0x10025c470 <+176>: jne    0x10025c483               ; <+195> at scriptnum_tests.cpp
    0x10025c472 <+178>: addq   $0x28, %rsp
    0x10025c476 <+182>: popq   %rbx
    0x10025c477 <+183>: popq   %r12
    0x10025c479 <+185>: popq   %r13
    0x10025c47b <+187>: popq   %r14
    0x10025c47d <+189>: popq   %r15
    0x10025c47f <+191>: popq   %rbp
    0x10025c480 <+192>: retq   
->  0x10025c481 <+193>: ud2    
    0x10025c483 <+195>: callq  0x100deea5c               ; symbol stub for: __stack_chk_fail
    0x10025c488 <+200>: nopl   (%rax,%rax)
```

**Comment by rex4539** [comment]
@defuse

**Comment by defuse** [comment]
This should be fixed by #4484

**Comment by rex4539** [comment]
I can confirm the patch fixes it.

**Comment by rex4539** [comment]
Fixed by https://github.com/zcash/zcash/pull/4484.

**Comment by daira** [comment]
Actually fixed by #4740 (which used a similar approach to #4484). Thanks @rex4539!
